apps/builder/src/features/editor/components/TypebotHeader/TypebotHeader.tsx

```diff
@@ -52,6 +52,39 @@ export const TypebotHeader = () => {
     setRightPanel(RightPanel.PREVIEW)
   }
 
+  const isUndoShortcut = (event: KeyboardEvent) => {
+    const key = event.key.toLowerCase()
+    const cmdHeld = event.metaKey
+    const ctrlHeld = event.ctrlKey
+    const Z = 'z'
+
+    return (cmdHeld && key === Z) || (ctrlHeld && key === Z)
+  }
+
+  useEffect(() => {
+    document.addEventListener(
+      'keydown',
+      (event: KeyboardEvent) => {
+        if (isUndoShortcut(event) && canUndo) {
+          undo()
+        }
+      },
+      false
+    )
+
+    return () => {
+      document.removeEventListener(
+        'keydown',
+        (event: KeyboardEvent) => {
+          if (isUndoShortcut(event) && canUndo) {
+            undo()
+          }
+        },
+        false
+      )
+    }
+  }, [canUndo, undo])
```

Comment:
You can use the `useEventListener` hook from chakra to avoid having to add a clean up function

Reply:
I have no experience using chakra, but would it be something like that?

`  useEventListener('keydown', (event: KeyboardEvent) => {
    if (isUndoShortcut(event) && canUndo) {
      undo()
    }
  }, document, { passive: false })`

Reply:
Any reason why you are setting passive to `false`?

```ts
useEventListener('keydown', (event: KeyboardEvent) => {
    if (isUndoShortcut(event) && canUndo) {
      undo()
    }
  })
```

That's maybe enough

Reply:
refactored